pkg/utils/utils.go

```diff
@@ -16,10 +23,37 @@ var (
 	NetDirectory = "/sys/class/net"
 	// SysBusPci is sysfs pci device directory
 	SysBusPci = "/sys/bus/pci/devices"
+	// SysV4ArpNotify is the sysfs IPv4 ARP Notify directory
+	SysV4ArpNotify = "/proc/sys/net/ipv4/conf/"
+	// SysV6NdiscNotify is the sysfs IPv6 Neighbor Discovery Notify directory
+	SysV6NdiscNotify = "/proc/sys/net/ipv6/conf/"
 	// UserspaceDrivers is a list of driver names that don't have netlink representation for their devices
 	UserspaceDrivers = []string{"vfio-pci", "uio_pci_generic", "igb_uio"}
 )
 
+// EnableArpAndNdiscNotify enables IPv4 arp_notify and IPv6 ndisc_notify for VF
+func EnableArpAndNdiscNotify(ifName string) error {
+	/* For arp_notify, when a value of "1" is set then a Gratuitous ARP request will be sent
+	 * when the network device is brought up or if the link-layer address changes.
+	 * For ndsic_notify, when a value of "1" is set then a Unsolicited Neighbor Advertisement
+	 * will be sent when the network device is brought up or if the link-layer address changes.
+	 * Both of these being enabled would be useful in the case when an application reenables
+	 * an interface or if the MAC address configuration is changed. The kernel is responsible
+	 * for sending of these packets when the conditions are met.
+	 */
+	v4ArpNotifyPath := filepath.Join(SysV4ArpNotify, ifName, "arp_notify")
+	err := ioutil.WriteFile(v4ArpNotifyPath, []byte("1"), os.ModeAppend)
+	if err != nil {
+		return fmt.Errorf("failed to write arp_notify=1 for interface %s: %v", ifName, err)
+	}
+	v6NdiscNotifyPath := filepath.Join(SysV6NdiscNotify, ifName, "ndisc_notify")
+	err = ioutil.WriteFile(v6NdiscNotifyPath, []byte("1"), os.ModeAppend)
```

Comment:
if ipv6 is disabled for interface, will the path exist ? 

e.g
`sudo sysctl -w net.ipv6.conf.eth0.disable_ipv6=1`
`net.ipv6.conf.all.disable_ipv6=1`

Reply:
it does :)

Reply:
Thanks for confirming that they exist in the case when ipv6 is disable :) !